syntax = "proto3";

package shorrent;

message Chunk {
  uint32 id = 1;
  string filename = 2;
  string path = 3;
  uint32 size = 4;

  enum ChunkState {
    inTransfer = 0;
    done = 1;
  };
  ChunkState state = 5;

  repeated string peers = 6;
}

message File {
  string filename = 1;
  uint32 size = 2;
  repeated Chunk chunks = 3;
}

message RegFile {
  string address = 1;
  repeated File files = 2;
}

message FileList {
  repeated File files = 1;
}

message RegChunk {
  string address = 1;
  string filename = 2;
  uint32 id = 3;
}

message Data {
  bytes data = 1;
}

message Operation {
  enum Type {
    regFile = 0;
    fileList = 1;
    getFileInfo = 2;
    regChunk = 3;
    ok = 4;
    getChunk = 5;
  };
  Type op = 1;
  bytes data = 2;
  string msg = 3;
}
